packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -723,10 +737,24 @@ abstract class FlutterCommand extends Command<void> {
     return null;
   }();
 
+  DeviceConnectionInterface? get deviceConnectionInterface  {
+    if ((argResults?.options.contains(FlutterOptions.kDeviceConnection) ?? false)
+        && (argResults?.wasParsed(FlutterOptions.kDeviceConnection) ?? false)) {
```

Comment:
Do you need to check this, can `DeviceConnectionInterface` become nonnull and default to `both`?

Reply:
I think having it be `null` makes more sense. This function returns a `DeviceConnectionInterface`, which only has two options.
https://github.com/flutter/flutter/blob/f699eb0be0a0cfcf34e2b6c0e402a18d76a96f2e/packages/flutter_tools/lib/src/device.dart#L596-L600
Now hypothetically, I could add a 3rd option for `both`. However, `DeviceConnectionInterface` is also an attribute of `Device` and I don't think it'd make sense for `both` to be an option in that use case. People might think `both` means it's connected via both usb and wireless.
https://github.com/flutter/flutter/blob/f699eb0be0a0cfcf34e2b6c0e402a18d76a96f2e/packages/flutter_tools/lib/src/device.dart#L626-L627
I could make it an entirely new type but that seems like it would complicate it more

Reply:
> I could add a 3rd option for `both`.

How about `any`?

Reply:
Hmm `any` is better, but still wouldn't want the device's `connectionInterface` attribute to be set to it. Would a comment suffice to explain it, though?

```
/// How a device is connected.
enum DeviceConnectionInterface {
  attached,
  wireless,

  /// Used for filtering devices by `DeviceConnectionInterface`. A device's
  /// `connectionInterface` should not be set to `any`.
  any,
}
```

Reply:
What you have now is fine.